### ISSUE TITLE
CommonJS Modules

### ISSUE THREAD

**Issue by richard-flosi** [RReverser/gulp-wrap-js#8]
Can I use `gulp-wrap-js` to register CommonJS Modules?

I'm trying to provide a template like:
`.pipe(wrapJS('require.register("???", function(exports, require, module){ %= body % });'))`
but I don't know how to pass in the module name where I have `???`.

Any suggestions?

**Comment by RReverser** [comment]
Yes, you can do `...require.register(<%= { type: 'Literal', value: "???" } %>, function(...`.

P.S. Real CommonJS doesn't have `require.register`, I guess you meant AMD or something else?

**Comment by richard-flosi** [comment]
I'm using https://github.com/efacilitation/commonjs-require to use `require()` in the browser.

I still can't figure out a way to get the current file path and pass a variable into the template successfully.

**Comment by jacobthemyth** [comment]
@richard-flosi Though not documented, you can access the file path in the template, so my code looks like:

``` js
.pipe(wrap('require.register("<%= file.relative.replace(\'.js\',\'\') %>", function(exports, require, module){ %= body % });'))
```

**Comment by richard-flosi** [comment]
Thanks @jacobthemyth
